File: PTOD24.py
#THIS CODE IS WRITTEN IN PYTHON for taday's PTOD streak

class Solution:
    def detectLoop(self, head):
        #code here
        s, f = head, head.next
        while f and f.next:
            if s == f:
                return True
            s = s.next
            f = f.next.next
        return False

File: PTOD24.java
//This code is written in java

class Solution {
    public static Node findFirstNode(Node head) {
        Node slow = head, fast = head;
        while (fast != null && fast.next != null) {
            if ((slow = slow.next) == (fast = fast.next.next)) {
                for (slow = head; slow != fast; slow = slow.next, fast = fast.next);
                return slow;
            }
        }
        return null;
    }
}
